tests/subscriptions/entity_subscriptions/test_entity_subscriptions.py

```diff
@@ -1,158 +1,296 @@
-    pytest.param(
-        EventsSubscription(data_dict={}),
-        [],
-        True,
-        id="Events subscription with offset of type SNQL",
-    ),
-    pytest.param(
-        EventsSubscription(data_dict={}),
-        [],
-        False,
-        id="Events subscription with no offset of type SNQL",
-    ),
-    pytest.param(
-        TransactionsSubscription(data_dict={}),
-        [],
-        True,
-        id="Transactions subscription with offset of type SNQL",
-    ),
-    pytest.param(
-        TransactionsSubscription(data_dict={}),
-        [],
-        False,
-        id="Transactions subscription with no offset of type SNQL",
-    ),
-    pytest.param(
-        SessionsSubscription(data_dict={"organization": 1}),
-        [
-            binary_condition(
-                ConditionFunctions.EQ,
-                Column(None, None, "org_id"),
-                Literal(None, 1),
+        EntityKey.METRICS_SETS,
+        Query(
+            QueryEntity(
+                EntityKey.METRICS_SETS,
+                get_entity(EntityKey.METRICS_SETS).get_data_model(),
             ),
-        ],
-        True,
-        id="Sessions subscription of type SNQL",
-    ),
-    pytest.param(
-        MetricsCountersSubscription(data_dict={"organization": 1}),
-        [
-            binary_condition(
-                ConditionFunctions.EQ,
-                Column(None, None, "org_id"),
+            selected_columns=[
+                SelectedExpression(
+                    "time", Column("_snuba_timestamp", None, "timestamp")
+                ),
+            ],
+            condition=binary_condition(
+                "equals",
+                Column("_snuba_project_id", None, "project_id"),
                 Literal(None, 1),
             ),
-        ],
-        True,
-        id="Metrics counters subscription of type SNQL",
+        ),
+        {"organization": 1},
+        None,
+        id="Metrics sets subscription",
     ),
     pytest.param(
-        MetricsSetsSubscription(data_dict={"organization": 1}),
-        [
-            binary_condition(
-                ConditionFunctions.EQ,
-                Column(None, None, "org_id"),
+        EntityKey.METRICS_SETS,
+        Query(
+            QueryEntity(
+                EntityKey.METRICS_SETS,
+                get_entity(EntityKey.METRICS_SETS).get_data_model(),
+            ),
+            selected_columns=[
+                SelectedExpression(
+                    "time", Column("_snuba_timestamp", None, "timestamp")
+                ),
+            ],
+            condition=binary_condition(
+                "equals",
+                Column("_snuba_project_id", None, "project_id"),
                 Literal(None, 1),
             ),
-        ],
-        True,
-        id="Metrics sets subscription of type SNQL",
+        ),
+        {},
+        InvalidQueryException,
+        id="Metrics sets subscription",
     ),
 ]
 
+# TESTS_CONDITIONS_SNQL_METHOD = [
```

Comment:
Why are these commented out?

Reply:
These were stale which were already captured by the new validator and processor tests. Removed.